Constructive Code Review/outline.txt

```diff
@@ -0,0 +1,205 @@
+              (If they don't put up their hands: "Of course you don't.
+              It's a secret.")
+        - Almost everybody does. It doesn't really go away. Realize
+          nobody's judging you; they're too busy worrying about people
+          judging them.
+        - Realize you've already made it.
+            - If you're hired, congratulations. We don't have an easy
+              hiring process. You're competent to do this job.
+            - Sometimes somebody will find a problem in your PR because
+              they know something you don't. Sometimes it's just having
+              a fresh set of eyes.
+        - Who's afraid their imposter syndrome isn't as good as
+          everybody else's?
+        - It's been a help to me to notice when I'm feeling defensive
+          or angry.
+            - Why am I feeling that way? Sometimes that reveals things
+              I believe that aren't true or things I need to think
+              about.
+            - I generally make bad decisions and say things I regret
+              when feeling that way.
+            - It's a good time to talk a break and come back with a
+              clear head.
+    - Feeling short on time
+          There will always be more to review than can be reviewed. So
+          we have to learn to adapt to that reality: lower standards,
+          never sleep, or just go at a sustainable pace. It can do a
+          lot for a contrib's peace of mind just to know you've seen
+          that the patch is there and that you haven't forgotten about
+          it. Sometimes I even let people know "It's next in my queue."
+        - Consider: what's the risk if there's something wrong with it?
+            - Will it cause a lot of damage?
+            - Will it be easy to fix, or is there, say, a data-exchange
+              format being defined?
+    - Interpersonal issues
+        - When you're in person with your team, make a point of
+          establishing good relations with them.
+            - Then when you have to critique each other or talk on a
+              low-bandwidth medium like IRC, you'll know how to take
+              each other. You'll have some built-up offense-buffering.
+    - Other feelings
+        - When you can't decide what to say (something feels too bad to
+          merge), you can always fall back to articulating your
+          emotions: "I feel uncomfortable about leaving plain-text
+          passwords on disk."
+        - You don't have to be definitive, decisive, or an authority
+          all the time. Just treat everybody like adults and articulate
+          your feelings. I've said "I'm happy you want to work on that,
+          but I'm afraid that you'll end up in merge hell if I land the
+          elasticsearch stuff first." Inviting them into the tradeoff
+          is a heckuva lot better than "no!". (If they don't mine merge
+          hell, they can go for it!)
+- Dealing with newbies
```

Comment:
Perfect. I was a little worried about that but couldn't think of a word.

> Is it that they're new to programming or new to the project in question or both?

New in whatever the relevant sense to the interaction is